### PR TITLE
feat: pass stack config through to pulumi providers

### PR DESCRIPTION
**Pull request by asalkeld**
I used a stack config of :
```yaml
name: staging
provider: gcp
region: australia-southeast1
project: thing
config:
  default:
    memory: 512
  functions/api/main.go:
    timeout: 30
```

### CODE REVIEW

File: pkg/project/options.go [hunk 1]
```diff
@@ -25,58 +25,81 @@ import (
...
 	if err != nil {
 		return Function{}, err
 	}
 
+	c, ok := s.Config[h]
+	if !ok {
+		return Function{
+			ComputeUnit: ComputeUnit{
+				Name:     rt.ContainerName(),
+				Memory:   128,
```
Comment (asalkeld) [inline pkg/project/options.go]: so one thing is our current gcp has a mem default of 512..

File: pkg/project/types.go [hunk 2]
```diff
@@ -39,14 +39,17 @@ type ComputeUnit struct {
 	// Triggers used to invoke this compute unit, e.g. Topic Subscriptions
 	Triggers Triggers `yaml:"triggers,omitempty"`
 
-	// The memory of the compute instance in MB
+	// The memory of the compute instance in MB: default 128
 	Memory int `yaml:"memory,omitempty"`
 
-	// The minimum number of instances to keep alive
+	// The minimum number of instances to keep alive: default 1
```
Comment (asalkeld) [inline pkg/project/types.go]: should be 0

File: pkg/project/options.go [hunk 3]
```diff
@@ -25,58 +25,81 @@ import (
...
-func FunctionFromHandler(h, stackDir string) (Function, error) {
+func FunctionFromHandler(h string, s *stack.Config) (Function, error) {
 	pterm.Debug.Println("Using function from " + h)
 
 	rt, err := runtime.NewRunTimeFromHandler(h)
 	if err != nil {
 		return Function{}, err
 	}
 
+	c, ok := s.Config[h]
```
Comment (asalkeld) [inline pkg/project/options.go:79]: this ^ is not smart and needs the full name of the handler.. could be improved, also i don't have the "default" case yet..

File: pkg/project/options.go (hunk 1 above)
Comment (asalkeld) [reply pkg/project/options.go]: I have now taken this into account

File: pkg/stack/types.go [hunk 4]
```diff
@@ -25,9 +27,30 @@ const (
...
+	}
+
+	switch provider {
+	case Gcp:
+		ret.Memory = to.IntPtr(512)
+	default:
+		ret.Memory = to.IntPtr(128)
+	}
+
+	return ret
```
Comment (tjholm) [inline pkg/stack/types.go]: Would it be possible to shift this config into the providers themselves so they can set their own defaults in the case where its not defined in the configuration?

**Review by tjholm — COMMENTED** [review]
LGTM just one comment.

@jyecusch was also looking at updating the defaults for AWS to get higher default vCPU allocation.

**Review by tjholm — APPROVED** [review]

**Pull request merged**